src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -223,6 +223,30 @@ public static unsafe void Clear(Array array, int index, int length)
             // GC.KeepAlive(array) not required. pMT kept alive via `ptr`
         }
 
+        private unsafe nint GetFlattenedIndex(int rawIndex)
```

Comment:
This can be simplified to:
```
        private unsafe nint GetFlattenedIndex(int rawIndex)
        {
            // Checked by the caller
            Debug.Assert(Rank == 1);

            if (RuntimeHelpers.GetMethodTable(this)->IsMultiDimensionalArray)
            {
                ref int bounds = ref RuntimeHelpers.GetMultiDimensionalArrayBounds(this);
                rawIndex -= Unsafe.Add(ref bounds, 1);
            }

            if ((uint)rawIndex >= (uint)LongLength)
                 ThrowHelper.ThrowIndexOutOfRangeException();
            return rawIndex;
        }
```

Reply:
Ah, yeah that's much better 😄
Done in 664eecd8f665ad41cb7208823cc141bbb0ef277a both on CoreCLR and NativeAOT.